src/sql/workbench/services/contextualization/common/serverContextualizationService.ts

```diff
@@ -63,32 +56,86 @@ export class ServerContextualizationService extends Disposable implements IServe
 		throw invalidProvider(providerId);
 	}
 
+	/**
+	 * Contextualizes the provided URI for GitHub Copilot.
+	 * @param uri The URI to contextualize for Copilot
+	 * @returns A promise that doesn't need to be awaited since contextualization is done in the background.
+	 */
+	public async contextualizeUriForCopilot(uri: string): Promise<void> {
+		// Don't need to take any actions if contextualization is not enabled and can return
+		const isContextualizationNeeded = await this.isContextualizationNeeded();
+		if (!isContextualizationNeeded) {
+			return;
+		}
+
+		const getServerContextualizationResult = await this.getServerContextualization(uri);
```

Comment:
If you're planning on consolidating into one message than this is fine for now, but if you're keeping both then there's some room for cleanup here since you get context from both the generate and get calls which is a bit redundant.